### PR TITLE
[v6r17] Fixes in ProcessPool, DownloadInputData in JobScheduling

### PR DESCRIPTION
**Pull request by phicharp**
* ProcessPool: fix the finalization

* DownloadInputData: metadata keys were used as Storage Elements in case of failure. Make the check... This would fix JIRA issue https://its.cern.ch/jira/browse/LHCBDIRAC-663

* JobScheduling: 
   - try and select replicas for staging at the same site as online files
   - put sites holding data before others in the list of available sites

### CODE REVIEW

**Comment by chaen** [comment]
Review OK with me, but the tests should be ran again after the PR fixing them has been merged

File: WorkloadManagementSystem/Client/DownloadInputData.py [hunk 1]
```diff
@@ -100,6 +102,10 @@ def execute( self, dataToResolve = None ):
       # Get and remove size and GUIS
       size = reps.pop( 'Size' )
       guid = reps.pop( 'GUID' )
+      # Remove all other items that are not SEs
+      for item in reps.keys():
```
Comment (fstagni) [inline WorkloadManagementSystem/Client/DownloadInputData.py]: no need for `.keys()`

File: StorageManagementSystem/Client/test/Test_Client_StorageManagementSystem.py [hunk 2]
```diff
@@ -37,6 +37,10 @@
                                                     'Failed':{}} )
 mockObjectSE5.getStatus.return_value = S_OK( {'DiskSE': True, 'TapeSE':False} )
 
+mockObjectDMSHelper = MagicMock()
+mockObjectDMSHelper.getLocalSiteForSE.return_value = S_OK( 'mySite' )
+mockObjectDMSHelper.getSitesForSE.return_value = S_OK( ['mySite'] )
```
Comment (fstagni) [inline StorageManagementSystem/Client/test/Test_Client_StorageManagementSystem.py]: Is this mock object ever used?

File: WorkloadManagementSystem/Client/DownloadInputData.py (hunk 1 above)
Comment (phicharp) [inline WorkloadManagementSystem/Client/DownloadInputData.py]: Are you sure? Watch the pop just below ;-)

File: StorageManagementSystem/Client/test/Test_Client_StorageManagementSystem.py (hunk 2 above)
Comment (phicharp) [inline StorageManagementSystem/Client/test/Test_Client_StorageManagementSystem.py]: yes, when selecting the best site for staging we try and choose SEs at the same site (this is new)

File: WorkloadManagementSystem/Client/DownloadInputData.py (hunk 1 above)
Comment (fstagni) [inline WorkloadManagementSystem/Client/DownloadInputData.py]: Ah right :)

**Comment by fstagni** [comment]
Review OK for me but I would like to see also a review from somebody else.

**Pull request merged**